core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -1555,7 +1559,7 @@ private void checkExponentialCnf(int n) {
             eq(aRef, literal(3)),
             or(eq(aRef, literal(3)),
                 eq(aRef, literal(4)))),
-        "AND(SEARCH(?0.b, Sarg[2]), SEARCH(?0.a, Sarg[3]))");
+        "AND(=(?0.b, 2), =(?0.a, 3))");
```

Comment:
It would be sad if Calcite's plans flip back and forth from `SEARCH(a, 2)` to `=(a, 2)`. Can Sarg be implemented in such a way that it does not result in constant plan changes?